lib/active_record/connection_adapters/postgresql_fallback_adapter.rb

```diff
@@ -0,0 +1,113 @@
+require 'active_record/connection_adapters/abstract_adapter'
+require 'active_record/connection_adapters/postgresql_adapter'
+require 'discourse'
+
+class PostgreSQLFallbackHandler
+  include Singleton
+
+  attr_reader :running
+  attr_accessor :master
+
+  def initialize
+    @master = true
+    @running = false
+  end
+
+  def verify_master
+    return if @running && recently_checked?
+    @running = true
+
+    Thread.new do
+      begin
+        logger.info "#{self.class}: Checking master server..."
+        connection = ActiveRecord::Base.postgresql_connection(config)
+
+        if connection.active?
+          logger.info "#{self.class}: Master server is active. Reconnecting..."
+          ActiveRecord::Base.establish_connection(config)
+          Discourse.disable_readonly_mode
+          @master = true
+        end
+      rescue => e
+        if e.message.include?("could not connect to server")
+          logger.info "#{self.class}: Connection to master PostgreSQL server failed with '#{e.message}'"
```

Comment:
I would warn here, a warning every 5 secs when db is on fire is not the end of the world